[CORE-8805] dt/archival: Decrease manifest upload interval to avoid race

oleiman opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

This applies to 'ArchivalTest.test_all_partitions_leadership_transfer', where
validation checks that the end offset of the topic matches the end of the
partition manifest in the bucket.

In some cases, the test NTP stabilizes just after a partition manifest upload,
which means that we will have to wait at least a minute before the manifest is
in sync. This in turn puts us over the validation timeout for the test.

So this commit cuts the manifest_max_upload_interval in half for this test
case only. The alternative is to increase the test timeout, but that would
increase the runtime of the test in these outlier cases.


This build is the closest we're going to get to "proof" imo:

https://ci-artifacts.dev.vectorized.cloud/redpanda/68786/0197f643-a862-4918-9188-ab9daefa74cb/vbuild/ducktape/results/2025-07-10--001/report.html

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.1.x
- [x] v24.3.x
- [x] v24.2.x

## Release Notes

* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

oleiman commented:
/ci-repeat 3
release
skip-units
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer
dt-repeat=30

oleiman commented:
/ci-repeat 3
release
skip-redpanda-build
skip-units
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer
dt-repeat=200

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--d2854109-d282-45b8-bc0f-d4be610045cd-->
<!--TEST RESULTS-->
<!--cf67d990-4313-4872-bd4e-05ba96507f51-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/68569">build#68569</a></summary>


| test_class   | test_method                             | test_arguments                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                    |
|:-------------|:----------------------------------------|:------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------|
| ArchivalTest | test_all_partitions_leadership_transfer | {"cloud_storage_type": 2, "timeout": 120} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68569#0197e886-36f0-4fae-841e-21882d917da3 | FLAKY         | 219/220  | The test was found to be new, and no failures are allowed |
| ArchivalTest | test_all_partitions_leadership_transfer | {"cloud_storage_type": 2, "timeout": 120} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68569#0197e886-36f1-408f-880c-bda894c3f669 | FLAKY         | 219/220  | The test was found to be new, and no failures are allowed |
| ArchivalTest | test_all_partitions_leadership_transfer | {"cloud_storage_type": 2, "timeout": 180} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68569#0197e886-36f1-408f-880c-bda894c3f669 | FLAKY         | 219/220  | The test was found to be new, and no failures are allowed |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/68786">build#68786</a></summary>


| test_class   | test_method                             | test_arguments                            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                    |
|:-------------|:----------------------------------------|:------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------|
| ArchivalTest | test_all_partitions_leadership_transfer | {"cloud_storage_type": 2, "timeout": 120} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68786#0197f643-a862-40da-86f8-d25e25f3f196 | FLAKY         | 219/220  | The test was found to be new, and no failures are allowed |
| ArchivalTest | test_all_partitions_leadership_transfer | {"cloud_storage_type": 2, "timeout": 120} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/68786#0197f643-a862-4918-9188-ab9daefa74cb | FLAKY         | 215/220  | The test was found to be new, and no failures are allowed |
</details>
<!--/body-->

oleiman commented:
/ci-repeat 3
release
skip-redpanda-build
skip-units
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer
dt-repeat=200

oleiman committed:
dt/archival: Decrease manifest upload interval to avoid race

This applies to 'ArchivalTest.test_all_partitions_leadership_transfer', where
validation checks that the end offset of the topic matches the end of the
partition manifest in the bucket.

In some cases, the test NTP stabilizes just after a partition manifest upload,
which means that we will have to wait at least a minute before the manifest is
in sync. This in turn puts us over the validation timeout for the test.

So this commit cuts the manifest_max_upload_interval in half for this test
case only. The alternative is to increase the test timeout, but that would
increase the runtime of the test in these outlier cases.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman commented:
/ci-repeat 3
release
skip-redpanda-build
skip-units
tests/rptest/tests/archival_test.py::ArchivalTest.test_all_partitions_leadership_transfer
dt-repeat=200

nvartolomei commented:
Is the correct fix here this https://github.com/redpanda-data/redpanda/pull/25781 by chance? Logically the manifest is dirty so it should be uploaded asap rather than on interval.

nvartolomei commented:
But then we also do new writes right after restarts.

```
self.kafka_tools.produce(self.topic, 5000, 1024)
```

So there are new uploads from the leader which should immediately upload manifest, right? Or there are uploads and a leadership change after TS uploads but before manifest? I.e. leader rebalancing?

oleiman commented:
@nvartolomei I think this is more a case of the tip of an ntp falling out of sync with the partition manifest without any additional segments closing or uploads occurring, which is normal if there are just a handful of writes since the last sync.

Lazin commented:
> So there are new uploads from the leader which should immediately upload manifest, right? Or there are uploads and a leadership change after TS uploads but before manifest? I.e. leader rebalancing?

The uploads are not always triggering the manifest reupload, the condition is upload + enough time have passed since the previous manifest upload. And the manifest which is being uploaded is usually one step behind the one in the local in-memory state of the STM.

vbotbuildovich commented:
/backport v25.1.x

vbotbuildovich commented:
/backport v24.3.x

vbotbuildovich commented:
/backport v24.2.x

nvartolomei commented:
Evgeny raised an important detail I missed. Any reason we are not tuning
manifest upload interval when fast uploads is true in SISettings by
default? Sounds like a good optimisation.

On Mon, 14 Jul 2025 at 23:50, vbotbuildovich ***@***.***>
wrote:

> *vbotbuildovich* left a comment (redpanda-data/redpanda#26725)
> <https://github.com/redpanda-data/redpanda/pull/26725#issuecomment-3071263324>
>
> /backport v24.2.x
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/redpanda-data/redpanda/pull/26725#issuecomment-3071263324>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEETWNQPTTVPWHIQ2FNHR33IQX3DAVCNFSM6AAAAACA75WNZOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANZRGI3DGMZSGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
